src/util-file.c

```diff
@@ -875,7 +915,24 @@ static int FileCloseFilePtr(File *ff, const uint8_t *data,
     if ((flags & FILE_TRUNCATED) || (ff->flags & FILE_HAS_GAPS)) {
         ff->state = FILE_STATE_TRUNCATED;
         SCLogDebug("flowfile state transitioned to FILE_STATE_TRUNCATED");
-
+        /* for hashing truncated files... */
+#ifdef HAVE_NSS
```

Comment:
perhaps this can be turned into a small util func and be called from both truncated and closed cases

Reply:
Agreed, in fact in the first PR version was coded as you say but I removed it for the sake of changing as less as possible